lib/internal/errors.js

```diff
@@ -18,15 +18,13 @@ function makeNodeError(Base) {
   return class NodeError extends Base {
     constructor(key, ...args) {
       super(message(key, args));
-      this[kCode] = key;
-    }
-
-    get name() {
-      return `${super.name} [${this[kCode]}]`;
-    }
-
-    get code() {
-      return this[kCode];
+      this[kCode] = this.code = key;
+      Object.defineProperty(this, 'name', {
+        configurable: true,
+        enumerable: false,
+        value: `${super.name} [${this[kCode]}]`,
+        writable: true
+      });
     }
```

Comment:
If you moved the `name` assignment to the prototype it would align with builtin errors
*(no-own name prop on error objects)*.
```js
Object.defineProperty(NodeError.prototype, 'name', {
  configurable: true,
  enumerable: false,
  value: `${Base.prototype.name} [${this[kCode]}]`,
  writable: true
});

return NodeError
```

Reply:
Following up on [my previous comment](https://github.com/nodejs/node/pull/15694#discussion_r142001138).
It looks like getters and a symbol was used to mimic built-in errors which have non-enumerable inherited properties *(no-own properties)*.

So the tweaks needed to the existing errors would be adding setters for `code` and `name` with `enumerable`, `configurable`, and `writable` of `true`. To avoid the current chattier `console.log` the `kCode` symbol assignment could be done with `Object.defineProperty` and made `enumerable` of `false`.

@Trott If you're up for it I could send a PR to your PR, since I'm kind of walking through the code as is and I've never had a commit merged to Node.

Reply:
> It looks like getters and a symbol was used to mimic built-in errors which have non-enumerable inherited properties (no-own properties).

Yep. That's exactly why

Reply:
> @Trott If you're up for it I could send a PR to your PR, since I'm kind of walking through the code as is and I've never had a commit merged to Node.

@jdalton Sure, you can do that or if you'd rather just take it over, you can open your own PR and I can close this one. Either way is fine by me.